migrations/archive/2019/20190731_outage_repair_groups.js

```diff
@@ -0,0 +1,82 @@
+/* eslint-disable no-console */
+const MIGRATION_NAME = '20190731_outage_repair_groups';
+import { model as Group } from '../../website/server/models/group';
+import { model as User } from '../../website/server/models/user';
+import forEach from 'lodash/forEach';
+
+const progressCount = 1000;
+let count = 0;
+
+async function updateGroup (group) {
+  count++;
+
+  const set = {};
+
+  if (group.type === 'guild') {
+    set.memberCount = await User.countDocuments({guilds: group._id});
+  } else {
+    set.memberCount = await User.countDocuments({party._id: group._id});
+  }
+
+  if (group.quest && group.quest.key && group.quest.members) {
+    forEach(group.quest.members, async (status, memberId) => {
```

Comment:
lodash's for each doesn't work as expected with promises or async. It'll go ahead without waiting. Since this is an object you can use a normal `for...of` where you can use await but it'll process everything in sequence not parallel. Otherwise you can convert `group.quest.members ` to an array and use .map + Promise.all

Reply:
argh, I keep making that mistake. >.< Someday I'll learn!